examples/ros/src/smarts_ros/scripts/ros_driver.py

```diff
@@ -303,7 +303,7 @@ def _agent_spec_callback(self, ros_agent_spec: AgentSpec):
                 ),
             ),
         )
-        with self._control_lock:
+        with self._reset_lock:
```

Comment:
It seems we generate a mission above outside of the lock. This means that the ordering of applying to self._agents_to_add is not deterministic if multiple `smarts_reset()` get called. It should be extremely unlikely to cause a problem unless there is already a problem of reset calls.

Reply:
Good point!  I guess this could happen if someone double-clicks the Reset button (in  a hypothetical co-simulation environment that might have one).  But in that case, the agents being added will be the same and it shouldn't matter.  It could also happen, with different agents, if there are multiple "co-operating" clients trying to add agents at roughly the same time, but I think in such a situation the goal of determinism may not matter.  So I agree, I don't think it's worth the overhead to try to add a fix for this very-unlikely situation yet.